[ADD] sale_third_discount - modulo que permite añadir el tercer descuento en ventas

agaldona opened the pull request:
[IMP] añadir permisos para segundo descuento + cambiar cálculo subtotal de las líneas de pedido de venta

Commit:
[IMP] añadir permisos para segundo descuento + cambiar cálculo subtotal de las líneas de pedido de venta
[ADD] sale_third_discount - modulo que permite añadir el tercer descuento en ventas

pedrobaeza commented:
Ainara, puesto que has puesto la posibilidad de por parámetro mostrar u ocultar el segundo y tercer descuento, creo que no merece la pena el tener un módulo extra, ya que de todas formas tienes que sobreescribir el cálculo del subtotal. Mételo todo en el pricelist_rules.

anajuaristi commented:
+1 Good point

agaldona commented:
Nunca me sale bien a la primera.. :sweat_smile: Katxis...
Puesto que va a ir lo del 3er descuento en el pricelist rules, tendría que hacer al igual que lo hizo oihane, la parte del tercer descuento en la regla tarifaria, en el product_pricelist_rules? O lo meto todo en sale_pricelist_rules y ya?

pedrobaeza commented:
Pues ya que estamos, como sé que a Ana le va a gustar, ponlo también en las reglas tarifarias y luego lo llevas a venta, pero siempre si se tiene parametrizado que se vean. Entonces, habrá que mover los parámetros también a ese módulo.

agaldona commented:
En realidad lo habia hecho de esa forma. El tema es que ahora debo dividir en dos módulos, en el product_pricelist_rules y sale_pricelist_rules, en el primero va la parte de las reglas tarifarias, y en el segundo el de ventas. Los grupos de permisos los he puesto en el primero y la configuración en el segundo, por que? pues porque la configuración de descuentos, por lo que he podido ver, se configura en el módulo de ventas, y el módulo product_pricelist_rules, no tiene dependencia a ventas. No se si así estaría correcto. Es decir, que este limitado por grupos por defecto, y si tienes instalado ventas lo puedes configurar por lo check de configuración.

pedrobaeza commented:
Me parece bien esa opción que comentas.

anajuaristi commented:
Pues sí!!! Por mi parte, siempre dije que el tercer descuento no debía ir en un módulo aparte, sino ya de base. 
Montones de mini-erps de estos que nos encantan ;P Tienen los 3 descuentos ya metidos y aún hay bastantes clientes que lo usan. Otro montón que efectivamente, ya no lo necesitan pero es perfecto que se pueda elegir por configuración.

Commit:
[DEL] Eliminar módulo sale_third_discount
[IMP] dividir las funcionalidades del módulo sale_third_discount en los módulos product_pricelist_rules y sale_pricelist_rules

agaldona commented:
Ya he eliminado el módulo y he incorporado sus funcionalidades en los dos módulos de pricelist rules que hemos comentado.

pedrobaeza commented on product_pricelist_rules/i18n/es.po:
```diff
@@ -123,7 +128,7 @@ msgstr "Tipo de tarifa"
 #: view:product.pricelist.item:product_pricelist_rules.pricelist_item_search_view
 msgid "Pricelist Version"
-msgstr "Versión de tarifa"
+msgstr "Version de tarifa"
```
Por qué quitas la tilde?

agaldona replied:
Creo que no he tocado eso, la verdad, puede ser por la exportación, que no me lo haya traido?

pedrobaeza commented on product_pricelist_rules/models/product_pricelist.py:
```diff
@@ -42,6 +42,8 @@ class PricelistItem(models.Model):
                             digits=dp.get_precision('Product Price'))
     discount2 = fields.Float('Discount 2 %',
```
Para que ocupen menos las columnas, debería ser 'Disc. 2 (%)' (y cambiar las traducciones acorde a ello)

pedrobaeza commented on product_pricelist_rules/models/product_pricelist.py:
```diff
@@ -42,6 +42,8 @@ class PricelistItem(models.Model):
                             digits=dp.get_precision('Product Price'))
     discount2 = fields.Float('Discount 2 %',
                              digits=dp.get_precision('Product Price'))
+    discount3 = fields.Float('Discount 3 %',
```
Ídem

pedrobaeza commented on product_pricelist_rules/security/product_pricelist_rules_security.xml:
```diff
@@ -0,0 +1,12 @@
+<openerp>
+    <data noupdate="0">
+        <record id="group_second_discount" model="res.groups">
+            <field name="name">Second Discount</field>
```
El segundo descuento debería implicar también ver el primero

agaldona replied:
Si, pero al no tener dependencia con sale, se lo añado en el otro módulo.

pedrobaeza replied:
Ah, OK

pedrobaeza commented on sale_pricelist_rules/models/sale.py:
```diff
@@ -55,26 +57,39 @@ def _calculate_subtotal(self):
-        self.price_subtotal = cur.round(taxes['total'])
+    def _amount_line(self, cr, uid, ids, field_name, arg, context=None):
+        res = {}
+        tax_obj = self.pool.get('account.tax')
```
Por qué lo vuelves a poner con API vieja?

agaldona replied:
Por lo que te comenté de que no cargaba bien el resultado, que por mucho que pasara la función de la api nueva, al guardar ejecutaba la vieja y cargaba el valor de la vieja.

pedrobaeza replied:
Pero si Oihane lo tenía en la nueva API, sí que le funcionaba.

agaldona replied:
Pues yo por mucho que lo probara, me seguía cargando la vieja. Si me calculaba el nuevo, pero no guardaba eso.

pedrobaeza replied:
@oihane, puedes ver tú con @agaldona qué es lo que puede ocurrir?

Commit:
[IMP] añadir abreviaturas + cambiar traducciones

Commit:
[IMP] cambiar sale_pricelist_rules.pot para las abreviaturas

Commit:
[FIX] traceback al cargar una línea de producto que no coincide con una regla tarifaria

Commit:
[IMP] Etiqueta descuentos Dto ISSUE #667

pedrobaeza commented:
Ainara, ponle por favor con el punto (Dto.), y para que cierre el issue, tienes que poner "Fixes #667" o "Closes #667".

Commit:
[IMP] Etiqueta descuentos Dto. Closes #667

pedrobaeza commented:
:+1:
